src/store/proposals/sagas/addProposal.test.ts

```diff
@@ -0,0 +1,557 @@
+  false
+);
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('add proposal sagas', () => {
+  const reducer = combineReducers(pick(rootReducer, [Reducers.Proposals]));
+  const initialState: DeepPartial<RootState> = {
+    [Reducers.Proposals]: {
+      ...new ProposalsState(),
+    },
+  };
+
+  describe('addProposal', () => {
+    const initialSteps =
+      initialState[Reducers.Proposals]?.addProposalCall?.steps;
+
+    const mockTx = {
+      hash: '0x01',
+      wait: jest.fn() as ContractTransaction['wait'],
+    } as ContractTransaction;
+
+    const mockTxReceipt = {
+      transactionHash: '0x01',
+    } as TransactionReceipt;
+
+    const successState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'success',
+          currentOperation: 'propose',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              tx: mockTx,
+              txReceipt: mockTxReceipt,
+            },
+          ],
+        },
+      },
+    };
+    const errorMsg = 'An unexpected error has occurred. Please try again';
+
+    const failureState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: errorMsg,
+            },
+          ],
+          currentOperation: 'propose',
+        },
+      },
+    };
+
+    const failureWalletNotConnectedState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: 'Wallet not connected',
+            },
+          ],
+        },
+      },
+    };
+
+    const proposalGovernorAdminPayload: ProposalsActions['addProposal'] = {
+      payload: {
+        SendProposalContract: GovernorTypes.GovernorAdmin,
+        Description: 'test description of proposal',
+        Values: [
+          {
+            Target: '0x03f4',
+            Value: '1000000000',
+            Signature: 'Signature',
+            Calldata: '0x616263640000',
+          },
+        ],
+      },
+      type: 'proposals/addProposal',
+    };
+
+    const getBasePath = () =>
+      expectSaga(addProposal, proposalGovernorAdminPayload)
+        .withReducer(reducer)
+        .withState(initialState)
+        .select(accountSelector);
+
+    it('happy path', async () => {
+      await getBasePath()
+        .provide([
+          [matchers.select(accountSelector), testAccount],
+          [matchers.select(governorAdminSelector), mockGovernorContract],
+          [
+            matchers.call(
```

Comment:
Maybe that would be a good place to use `call.fn` to make the mock clearer?

```suggestion
            [matchers.call.fn(mockGovernorContract.propose), mockTx]
```

The difference is that `call.fn` does not require arguments and mocks all function calls. In your situation, since the `propose` method it only being called once, it should not make any difference.